File: app/src/main/java/worker/WorkerUtils.java
package worker;

import androidx.work.BackoffPolicy;
import androidx.work.Constraints;
import androidx.work.Data;
import androidx.work.NetworkType;
import androidx.work.OneTimeWorkRequest;
import androidx.work.Worker;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;
import java.util.concurrent.TimeUnit;

import model.Game;

public class WorkerUtils {
    public static OneTimeWorkRequest addDataToRequest(String dataToBeAdded) {
        return new OneTimeWorkRequest.Builder(AdacaWorker.class)
                .setConstraints(new Constraints.Builder()
                        .setRequiredNetworkType(NetworkType.CONNECTED)
                        .setRequiresBatteryNotLow(true)
                        .build())
                .setInputData(new Data.Builder()
                        .putString("game", dataToBeAdded)
                        .build())
                .setBackoffCriteria(
                        BackoffPolicy.LINEAR,
                        OneTimeWorkRequest.MIN_BACKOFF_MILLIS,
                        TimeUnit.MILLISECONDS)
                .build();
    }
}

File: app/src/main/kotlin/worker/WorkerUtils.kt
package worker

import androidx.work.*
import java.util.concurrent.TimeUnit

object WorkerUtils {
    fun addDataToRequest(dataToBeAdded: String): OneTimeWorkRequest {
        return OneTimeWorkRequest.Builder(AdacaWorker::class.java)
                .setConstraints(Constraints.Builder()
                        .setRequiredNetworkType(NetworkType.CONNECTED)
                        .setRequiresBatteryNotLow(true)
                        .build())
                .setInputData(Data.Builder()
                        .putString("game", dataToBeAdded)
                        .build())
                .setBackoffCriteria(
                        BackoffPolicy.LINEAR,
                        OneTimeWorkRequest.MAX_BACKOFF_MILLIS,
                        TimeUnit.MILLISECONDS

                )
                .build()
    }
}